File: regions/system/server/go/bff-proxy/internal/handler/health.go
package handler

import (
	"net/http"

	"github.com/gin-gonic/gin"
	"github.com/redis/go-redis/v9"
)

// HealthHandler provides liveness and readiness probes.
type HealthHandler struct {
	redisClient redis.Cmdable
}

// NewHealthHandler creates a new HealthHandler.
func NewHealthHandler(redisClient redis.Cmdable) *HealthHandler {
	return &HealthHandler{redisClient: redisClient}
}

// Healthz is the liveness probe. Returns 200 if the process is alive.
func (h *HealthHandler) Healthz(c *gin.Context) {
	c.JSON(http.StatusOK, gin.H{"status": "ok"})
}

// Readyz is the readiness probe. Checks Redis connectivity.
func (h *HealthHandler) Readyz(c *gin.Context) {
	if err := h.redisClient.Ping(c.Request.Context()).Err(); err != nil {
		c.JSON(http.StatusServiceUnavailable, gin.H{
			"status": "not_ready",
			"reason": "redis connection failed",
		})
		return
	}
	c.JSON(http.StatusOK, gin.H{"status": "ready"})
}

File: regions/system/server/rust/ai-gateway/src/adapter/handler/health.rs
// ヘルスチェックハンドラー。
// サーバーの稼働状態と準備状態を返す。

use axum::response::IntoResponse;
use axum::Json;

/// ライブネスプローブ。サーバーが稼働中であることを返す。
pub async fn healthz() -> impl IntoResponse {
    Json(serde_json::json!({"status": "ok"}))
}

/// レディネスプローブ。サーバーがリクエストを受け付ける準備ができていることを返す。
pub async fn readyz() -> impl IntoResponse {
    Json(serde_json::json!({"status": "ready"}))
}
